REST: Normalize URI for requests (force US-ASCII syntax)

matthiaso opened the pull request:
Starting with Apache HTTP client 5 the URI was not re-encoded/normalized
 by the HTTP client anymore (previously the HTTP client did this); see
 migration guide to HttpClient 5.x classic APIs.

matthiaso committed:
REST: Normalize URI for requests (force US-ASCII syntax)

Starting with Apache HTTP client 5 the URI was not re-encoded/normalized
by the HTTP client anymore (previously the HTTP client did this); see
migration guide to HttpClient 5.x classic APIs.

435113

paolobazzi commented on org.eclipse.scout.rt.rest.jersey.client/src/main/java/org/eclipse/scout/rt/rest/jersey/client/ScoutApacheConnector.java:
```diff
@@ -855,4 +864,22 @@ public String getKey() {
+
+    @Override
+    public String description() {
+      return "Specifies whether the URI should be re-encoded to force US-ASCII encoding (RFC 2396 compliant)";
```
Indicate that the default value is true

matthiaso replied:
See https://github.com/eclipse-scout/scout.rt/compare/08d130fbec9b9edda3b767d7add7aaba7338cc73..731f385a5361637c93be984bc29be2cd0f405609

paolobazzi commented on org.eclipse.scout.rt.rest.jersey.client/src/main/java/org/eclipse/scout/rt/rest/jersey/client/ScoutApacheConnector.java:
```diff
@@ -486,7 +488,14 @@ protected HttpUriRequestBase getUriHttpRequest(final ClientRequest clientRequest
-    HttpUriRequestBase request = new HttpUriRequestBase(clientRequest.getMethod(), clientRequest.getUri());
+    HttpUriRequestBase request = null;
+    try {
+      boolean encodeUriForRequest = CONFIG.getPropertyValue(RestForceEncodeUriForRequestProperty.class);
```
Add unit test with/without property set.
See org.eclipse.scout.rt.rest.jersey.client.ScoutApacheConnectorTest

matthiaso replied:
See https://github.com/eclipse-scout/scout.rt/compare/08d130fbec9b9edda3b767d7add7aaba7338cc73..731f385a5361637c93be984bc29be2cd0f405609

paolobazzi reviewed (commented):
Looks good, see inputs for minor improvements.

paolobazzi commented:
[EDIT] Remove ticket no. from commit message

matthiaso commented:
See https://github.com/eclipse-scout/scout.rt/compare/08d130fbec9b9edda3b767d7add7aaba7338cc73..731f385a5361637c93be984bc29be2cd0f405609 for changes since last review
